depguard is enabled and fails on v1.53.0

### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

Running v1.53.0 of golangci-lint seems to enable `depguard` and fails due to a config error. 

### Version of golangci-lint

<details>

```console
$ golangci-lint --version

v1.53.0
```

</details>

### Configuration file

<details></details>

### Go environment

<details>

```console
$ go version && go env

go version go1.20.4 darwin/amd64
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v

golangci/golangci-lint info checking GitHub for tag 'v1.53.0'
golangci/golangci-lint info found version: 1.53.0 for v1.53.0/darwin/amd64
golangci/golangci-lint info installed ./bin/golangci-lint
STARTING golangci-lint checks
ERRO [linter] depguard: create analyzer: couldn't expand $gostd: could not read GOROOT directory: open src: no such file or directory
must have an Allow and/or Deny package list 
```

</details>

### Code example or link to a public repository

Updating golangci-lint to v1.53.0 in https://github.com/moov-io/infra/blob/master/go/lint-project.sh 

CI run: https://github.com/moov-io/infra/actions/runs/5149020788/jobs/9271486191

```console
$ git clone git@github.com:moov-io/infra.git  
Cloning into 'infra'...
remote: Enumerating objects: 9277, done.
remote: Counting objects: 100% (96/96), done.
remote: Compressing objects: 100% (60/60), done.
remote: Total 9277 (delta 42), reused 86 (delta 36), pack-reused 9181
Receiving objects: 100% (9277/9277), 3.93 MiB | 1.45 MiB/s, done.
Resolving deltas: 100% (5751/5751), done.
$ cd infra      
$ golangci-lint run  
$
```

```console
$ git clone git@github.com:go-co-op/gocron.git                  
Cloning into 'gocron'...
remote: Enumerating objects: 1746, done.
remote: Counting objects: 100% (314/314), done.
remote: Compressing objects: 100% (161/161), done.
remote: Total 1746 (delta 218), reused 234 (delta 149), pack-reused 1432
Receiving objects: 100% (1746/1746), 592.91 KiB | 1.54 MiB/s, done.
Resolving deltas: 100% (1158/1158), done.
$ cd gocron
$ golangci-lint run
scheduler.go:1178:11: superfluous-else: if block ends with a break statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary) (revive)
} else {
repeatMap[dayOfMonth]++
}
scheduler_test.go:26:24: unused-parameter: parameter 'i' seems to be unused, consider removing or renaming it as _ (revive)
func (f fakeTime) Unix(i int64, i2 int64) time.Time {
^
scheduler_test.go:2356:54: unused-parameter: parameter 't' seems to be unused, consider removing or renaming it as _ (revive)
func TestScheduler_CheckSetBehaviourBeforeJobCreated(t *testing.T) {
^
```

I'm on Linux amd64.

--- a/go.mod
+++ b/go.mod
@@ -12,7 +12,6 @@ require (
 	github.com/BurntSushi/toml v1.3.0
 	github.com/Djarvur/go-err113 v0.0.0-20210108212216-aea10b59be24
 	github.com/GaijinEntertainment/go-exhaustruct/v2 v2.3.0
-	github.com/OpenPeeDeeP/depguard/v2 v2.0.1
 	github.com/alexkohler/nakedret/v2 v2.0.1
 	github.com/alexkohler/prealloc v1.0.0
 	github.com/alingse/asasalint v0.0.11
@@ -37,6 +36,7 @@ require (
 	github.com/go-xmlfmt/xmlfmt v1.1.2
 	github.com/gofrs/flock v0.8.1
 	github.com/golangci/check v0.0.0-20180506172741-cfe4005ccda2
+	github.com/golangci/depguard/v2 v2.0.2-0.20230601235138-ed68d3771f48
 	github.com/golangci/dupl v0.0.0-20180902072040-3e9179ac440a
 	github.com/golangci/go-misc v0.0.0-20220329215616-d24fe342adfe
 	github.com/golangci/gofmt v0.0.0-20220901101216-f2edd75033f2

--- a/pkg/golinters/depguard.go
+++ b/pkg/golinters/depguard.go
@@ -1,7 +1,7 @@
 package golinters
 
 import (
-	"github.com/OpenPeeDeeP/depguard/v2"
+	"github.com/golangci/depguard/v2"
 	"golang.org/x/tools/go/analysis"
 
 	"github.com/golangci/golangci-lint/pkg/config"

